pandas/tests/test_internals.py

```diff
@@ -655,7 +655,7 @@ def test_interleave(self):
         mgr = create_mgr('a: f8; b: i8')
         self.assertEqual(mgr.as_matrix().dtype, 'f8')
         mgr = create_mgr('a: f4; b: i8')
-        self.assertEqual(mgr.as_matrix().dtype, 'f4')
+        self.assertEqual(mgr.as_matrix().dtype, 'f8')
```

Comment:
oh sorry, I thought this was ``f4 f8`` (which we want to not upcast.

ok, yeah I think if you have integers then you should just force the upcast (for i8), yeah I suppose you can get more creative esp if you have i2 or something, but prob not worth the effort. I would do what numpy does for these numeric castings (e.g. use ``common_type``)